docs/designers-developers/glossary.md

```diff
@@ -19,6 +22,12 @@
 <dt>Block name</dt>
 <dd>A unique identifier for a block type, consisting of a plugin-specific namespace and a short label describing the block's intent. e.g. <code>core/image</code></dd>
 
+<dt>Block Template</dt>
+<dd>Templates are HTML files of block markup that map to templates from the standard WordPress template hierarchy, for example index, single or archive. This helps control the front-end defaults of a site that are not edited via the Page Editor or the Post Editor. </dd>
+
+<dt>Block Template Part</dt>
+<dd>Building on Block Templates, these parts help set structure like a Footer or Header that one typically sees in a WordPress site. With Full Site Editing and block based themes, users can create their own arbitrary Template Parts, save those in the database for their site, and re-use them throughout their site.</dd>
```

Comment:
I'd note that the key here is that template parts are generally _reusable_ parts like a header or footer.